server/infinispan/src/main/java/org/hawkular/apm/server/infinispan/InfinispanSpanCache.java

```diff
@@ -136,14 +147,8 @@ public void store(String tenantId, List<Span> spans, Function<Span, String> cach
             throw new NullPointerException("Id should not be null!");
         }
 
-        QueryFactory<?> queryFactory = Search.getQueryFactory(spansCache);
-        Query query = queryFactory.from(Span.class)
-                .having("parentId")
-                .eq(id)
-                .toBuilder().build();
+        Set<Span> children = childrenCache.get(id);
 
-        return query.<Span>list().stream()
-                .filter(span -> !span.serverSpan())
-                .collect(Collectors.toList());
+        return children == null ? null : new ArrayList<>(children);
```

Comment:
Same issue as with trace cache - if being used as a list for efficiency (iteration) then may be better to cache as a list rather than converting from set to list every time it is accessed.